src/foam/dao/EasyDAO.js

```diff
@@ -862,18 +823,31 @@ return this;
     {
       name: 'addIndex',
       type: 'foam.dao.EasyDAO',
+      documentation: 'Only relavent if the cache is true or if daoType was set to MDAO, but harmless otherwise. Adds an existing index to the MDAO',
       // TODO: The java Index interface conflicts with the js CLASS Index
       args: [ { javaType: 'foam.dao.index.Index', name: 'index' } ],
       code: function addIndex(index) {
         this.mdao && this.mdao.addIndex.apply(this.mdao, arguments);
         return this;
       },
       javaCode: `
-if ( getMdao() != null ) {
-  getMdao().addIndex(index);
-}
-return this;
-`
+          if ( getMdao() != null ) {
+            getMdao().addIndex(index);
+          }
+        return this;
+      `
     },
+    {
+      name: 'printDecorators',
```

Comment:
Where does this get called from? Note that we also have a script called "Show Decorators" that will do this.

Reply:
the dao itself; since EasyDAO is the end decorator for all daos once we migrate you just have to do a dao.printDecorators() either in a service script or anywhere in the source code for debugging